sympy/solvers/solveset.py

```diff
@@ -370,12 +370,13 @@ def _invert_complex(f, g_ys, symbol):
             g_ys_vars_1 = (k,) + g_ys_vars
             exp_invs = Union(*[imageset(Lambda((g_ys_vars_1,), (I*(2*k*pi + arg(g_ys_expr))
                                          + log(Abs(g_ys_expr)))), S.Integers**(len(g_ys_vars_1)))])
+            return _invert_complex(f.exp, exp_invs, symbol)
 
         elif isinstance(g_ys, FiniteSet):
             exp_invs = Union(*[imageset(Lambda(n, I*(2*n*pi + arg(g_y)) +
                                                log(Abs(g_y))), S.Integers)
                                for g_y in g_ys if g_y != 0])
-        return _invert_complex(f.exp, exp_invs, symbol)
+            return _invert_complex(f.exp, exp_invs, symbol)
```

Comment:
An alternative is to just check for `elif g_ys is S.EmptySet` and set `exp_invs = S.EmptySet` here, no other changes/improvements.